go-controller/pkg/node/node.go

```diff
@@ -478,6 +501,62 @@ func (n *OvnNode) Start(wg *sync.WaitGroup) error {
 	return err
 }
 
+//AddOrRemoveTaint adds or removes the network-unavailable taint on the node depending on the setTaint value
+func (n *OvnNode) AddOrRemoveTaint(setTaint bool) error {
+	action := ""
+	if setTaint {
+		action = "adding"
+	} else {
+		action = "removing"
+	}
+
+	node, err := n.Kube.GetNode(n.name)
+	networkUnavailableTaint := &kapi.Taint{Key: "k8s.ovn.org/network-unavailable", Value: "NoSchedule", Effect: "NoSchedule"}
+
+	if err != nil {
```

Comment:
nit: put error checking directly after `GetNode` and before assigning the `networkUnavailableTaint`